### PR TITLE
Add invocation checking and usage information in sumstyle and showstyle files

### PR DESCRIPTION
**Pull request by rafailagln**
This PR adds invocation checking and usage information in sumstyle and showstyle files.
1.  Check if `qmcalc` is installed
2. Add `-h` argument with usage message
3. Check if target directory exists.

### CODE REVIEW

File: src/showstyle.sh [hunk 1]
```diff
@@ -2,6 +2,18 @@
 #
 # Print the formatting style metrics for the specified directory
 #
+if ! [ -x "$(command -v qmcalc)" ]; then
```
Comment (dspinellis) [inline src/showstyle.sh]: the `command` built-in is AFAIK a *bash* feature.  Simply testing a null command invocation is simpler and more comprehensive: `if qmcalc </dev/null >/dev/null ...`

File: src/showstyle.sh [hunk 2]
```diff
@@ -2,6 +2,18 @@
...
+  echo 'Error: qmcalc is not installed.' >&2
+  exit 1
+fi
+if [ "$1" = "-h" ] || [ $# -eq 0 ]; then
+  echo "Usage: `basename $0` <target directory> -- reports the metrics for each C file in the directory"
+  exit 0
+fi
+if [ ! -d "$1" ]; then
```
Comment (dspinellis) [inline src/showstyle.sh:13]: Nice!

File: src/showstyle.sh [hunk 3]
```diff
@@ -2,6 +2,18 @@
 #
 # Print the formatting style metrics for the specified directory
 #
+if ! [ -x "$(command -v qmcalc)" ]; then
+  echo 'Error: qmcalc is not installed.' >&2
+  exit 1
+fi
+if [ "$1" = "-h" ] || [ $# -eq 0 ]; then
+  echo "Usage: `basename $0` <target directory> -- reports the metrics for each C file in the directory"
```
Comment (dspinellis) [inline src/showstyle.sh]: You need to add the `-h` option in the usage information.

File: src/sumstyle.sh [hunk 4]
```diff
@@ -2,6 +2,22 @@
 #
 # Summarize the formatting style metrics for the specified directory
 #
+if ! [ -x "$(command -v qmcalc)" ]; then
+  echo 'Error: qmcalc is not installed.' >&2
+  exit 1
+fi
+if [ "$1" = "-h" ] || [ $# -eq 0 ]; then
+  echo "Usage: `basename $0` <target directory> -- processes all C files in the directory
```
Comment (dspinellis) [inline src/sumstyle.sh]: Output description belongs to a manual page, not usage information. This is not a big deal though.

File: README.md [hunk 5]
```diff
@@ -40,6 +40,10 @@ This will also generate the files `header.tab` and `header.txt`,
 which can be used for loading the output into other programs
 for further processing.
 
+### Install
```
Comment (dspinellis) [inline README.md]: This commit was part of another pull request.  Please remove it from this pull request (e.g. with `rebase -i` and then `push -f`).

**Review by dspinellis — CHANGES_REQUESTED** [review]
Overall the PR shows developing maturity in shell scripting.

File: src/sumstyle.sh (hunk 4 above)
Comment (Rafaila98) [reply src/sumstyle.sh]: done

File: README.md (hunk 5 above)
Comment (rafailagln) [reply README.md]: Done

**Comment by dspinellis** [comment]
Thank you!

**Pull request merged**